### PR TITLE
feat(gitIntegration)

### PR DESCRIPTION
**Pull request by printf-twinkle**
### Functionality:
Integration with Github

### Solution:
Github Repo and Pull Request can be linked.

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
Go to MyOrganisation and click on Integrate with Github button to link the repo and a particular task to link the PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "@angular/platform-browser": "~13.0.0",
     "@angular/platform-browser-dynamic": "~13.0.0",
     "@angular/router": "~13.0.0",
+    "@octokit/core": "^3.5.1",
```
Comment (viv9k) [inline package.json:25]: Not required can be removed

File: src/app/body/task-details/task-details.component.ts [hunk 2]
```diff
@@ -325,4 +331,22 @@ export class TaskDetailsComponent implements OnInit {
       complete: () => console.info('Successful')
   });
   }
-}
\ No newline at end of file
+
+  // getPullRequests() {
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:335]: Remove commented code

File: src/app/services/http-service.service.ts [hunk 3]
```diff
@@ -12,7 +12,28 @@ export class HttpServiceService {
...
+  getPrDetails(){
+    const url = environment.githubApiUrl + "/repos/Worktez/worktez/pulls";
+    return this.httpClient.get(url);
+  }
+
   getReleaseDetails(){
     return this.httpClient.get(this.gitReleaseApiUrl);
   }
+
+  getGithubUserDetails() {
```
Comment (viv9k) [inline src/app/services/http-service.service.ts:24]: Why are you entering your username here, and if this method is not used we can make good removal of it.

File: src/environments/environment.prod.ts [hunk 4]
```diff
@@ -4,5 +4,6 @@ export const environment = {
     
   },
   useEmulators: false,
-  gitApiUrl: "https://api.github.com/repos/Worktez/worktez"
+  gitApiUrl: "https://api.github.com/repos/Worktez/worktez",
```
Comment (viv9k) [inline src/environments/environment.prod.ts:7]: Add the new variable in normal key as well

**Review by viv9k — CHANGES_REQUESTED** [review]
Package lock is having too many changes please revert it as it's not needed and complete other fixes as well

File: package.json (hunk 1 above)
Comment (printf-twinkle) [reply package.json:25]: done

File: src/app/body/task-details/task-details.component.ts (hunk 2 above)
Comment (printf-twinkle) [reply src/app/body/task-details/task-details.component.ts:335]: done

File: src/app/services/http-service.service.ts (hunk 3 above)
Comment (printf-twinkle) [reply src/app/services/http-service.service.ts:24]: done

File: src/environments/environment.prod.ts (hunk 4 above)
Comment (printf-twinkle) [reply src/environments/environment.prod.ts:7]: done

File: src/app/body/task-details/task-details.component.ts [hunk 5]
```diff
@@ -325,4 +331,12 @@ export class TaskDetailsComponent implements OnInit {
       complete: () => console.info('Successful')
   });
   }
-}
\ No newline at end of file
+
+  linkPr() {
+    this.gitPrEnabled = true;
+    console.log("clicked for github link")
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:337]: Remove this

**Review by viv9k — CHANGES_REQUESTED** [review]
Please do the required changes

File: src/app/body/task-details/task-details.component.ts (hunk 5 above)
Comment (viv9k) [reply src/app/body/task-details/task-details.component.ts:337]: Remove console log, everywhere.

File: functions/model/teams/teams.js [hunk 6]
```diff
@@ -95,6 +96,12 @@ fastify.post("/updateTeam", (req, res) => {
   updateTeam(req, res);
 });
 
+fastify.post("/addProjLink", (req, res) => {
+  console.log("here");
```
[the remark this answers is not in the document]
Comment (viv9k) [reply functions/model/teams/teams.js:100]: Remove console log

File: src/app/body/task-details/task-details.component.ts (hunk 5 above)
Comment (viv9k) [reply src/app/body/task-details/task-details.component.ts:337]: Also remove package-lock.json changes

File: functions/model/teams/teams.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (printf-twinkle) [reply functions/model/teams/teams.js:100]: done

File: src/app/body/task-details/task-details.component.ts (hunk 5 above)
Comment (printf-twinkle) [reply src/app/body/task-details/task-details.component.ts:337]: done

**Comment by viv9k** [comment]
revert package-lock file

**Comment by Pendyala-thanvi** [comment]
@printf-twinkle please revert package-lock.json file.

**Pull request merged**